aries_cloudagent/messaging/agent_message.py

```diff
@@ -98,6 +101,18 @@ def _get_handler_class(cls):
         """
         return resolve_class(cls.Meta.handler_class, cls)
 
+    @classmethod
+    def assign_version_to_message_type(cls, version: str):
```

Comment:
It seems like a real issue to me that the message type is being assigned at the class level and not the object level. It looks like instances of records like the OOB invitation may have a different version values, but the class is only loaded once and the version can only be set once until the agent is restarted?

Reply:
I have reworked it to address this issue.